apps/src/code-studio/progressReduxSelectors.js

```diff
@@ -302,11 +298,33 @@ export const getCurrentScriptLevelId = state => {
 };
 
 /**
- * Get the next level ID in the progression if it exists.
- * Returns undefined if not currently in a script level or
- * currently on the last level.
+ * Get a reference to the parent level, if it exists.
  */
-export const nextLevelId = state => {
+export const getParentLevel = state => {
+  const currentLevel = getCurrentLevel(state);
+
+  if (currentLevel?.parentLevelId) {
+    return levelById(
+      state.progress,
+      state.progress.currentLessonId,
+      currentLevel.parentLevelId
+    );
+  }
+
+  return;
+};
+
+/**
+ * Get a reference to the next level in the progression, if it exists.
+ *
+ * Bubble choice levels return the parent level in the case where the navigation
+ * type is not 'NEXT_LEVEL', since that is where the progression would take them
+ * when they press 'continue' in that case.
```

Comment:
Could you add in the doc what happens when type is `NEXT_LEVEL` and not `PARENT`? And to confirm my understanding, would the level after the parent level be returned? If so, would 'Continue to X' be displayed?

Reply:
Also, what happens if the `bubble_choice` level is last in the progression? Would 'Finish' be displayed?

Reply:
Tagging @sanchitmalhotra126 who added `navigation_type` in https://github.com/code-dot-org/code-dot-org/pull/69083 - it seems like if the sublevel navigation type is `NEXT_LEVEL`, then the level after parent level would be returned, and if the parent level was the last in a progression, `undefined` would be returned?